open-metadata-resources/open-metadata-samples/open-metadata-security-samples/src/main/java/org/odpi/openmetadata/metadatasecurity/samples/CocoPharmaServerSecurityProvider.java

```diff
@@ -18,7 +18,7 @@ public CocoPharmaServerSecurityProvider()
     {
         super();
 
-        Class<?>        connectorClass = CocoPharmaServerSecurityConnector.class;
+        Class<?>        connectorClass = AbstractCocoPharmaServerSecurityConnector.class;
```

Comment:
I know this is still work in progress but here are some hints.

The connector class can not be an abstract class because this is the name of the class that is used to create a new bean.

There are two approaches to designing the two modes of operation for this connector.
* Have two independent connectors - in which case they each need their own connector provider
* Have a single connector class that has the business rules in it.  When it needs to look up which groups a user is in, it has an interface to the user data.  There are two implementations of this interface - one for the in-memory user maps and one for LDAP.  The connector chooses which one it uses by a property in the connection's configuration properties.

Reply:
Hi Mandy! 

I'm very grateful for your input! I've been working with Nigel & David trying to develop this connector and I heard you spoke on connectors during the last call. I'll check out that call as I did need some guidance on how to go about completing these connector(s). 

I am going to keep your suggestion in mind, specifically the one about creating two connectors. As of right now we're trying to establish full functionality of the LDAP calls & such. Once that is done I'll be moving onto the connector side. 

I'll be posting a couple more of these drafts before I request a merge as there's plenty of testing that needs to be done.

Forgive my sluggishness, this is all very new to me but I'm very dedicated towards creating something useful for you guys.

I'll work more on this, expect a new PR soon!